src/lib/rest/rest.cpp

```diff
@@ -974,46 +993,56 @@ static int connectionTreat
   //
   if (dataLen != 0)
   {
-    if (dataLen == ciP->httpHeaders.contentLength)
+    //
+    // If the HTTP header says the request is bigger than our PAYLOAD_MAX_SIZE,
+    // just silently "eat" the entire message
+    //
+    if (ciP->httpHeaders.contentLength > PAYLOAD_MAX_SIZE)
```

Comment:
And it is not a bad question.
Unfortunately MHD doesn't let you write on the socket until the entire message is read.
This is pretty strange, as the socket is full duplex, but tests have been made and an error is returned if you try to write on the socket before it 'is ready'.

We could always close the socket (which is not a bad option), but I don't think we'd be able to return any error response.

Perhaps the best option here is just to **close** the socket ... ?